compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt

```diff
@@ -528,3 +530,16 @@ internal fun KotlinType.convertKotlinType(): KotlinType {
         upperBound
     )
 }
+
+private fun ClassifierDescriptor.shouldMapToKotlinForAssignabilityCheck(): Boolean {
```

Comment:
you can make it extension of nullable type to avoid null check before calling this. Alternatively, you can use `?.foo() == true`

Reply:
I did consider this, but then the smart cast for the `classifierDescriptor` will not be possible and a redundant `!!` will be required.

Reply:
Changed to `declarationDescriptor?.shouldMapToKotlinForAssignabilityCheck() == true`